robots/gazebo_tank.h

```diff
@@ -21,92 +21,39 @@ class GazeboTank
     using meters_per_second_t = units::velocity::meters_per_second_t;
 
 public:
-    GazeboTank(const meters_per_second_t maximumSpeed, const millimeter_t axisLength)
+    GazeboTank(const meters_per_second_t maximumSpeed)
       : m_MaximumSpeed(maximumSpeed)
-      , m_AxisLength(axisLength)
     {}
 
-    virtual millimeter_t getRobotWidth() const override
-    {
-        return m_AxisLength;
-    }
-
-    template<typename ReturnLengthUnit = LengthUnit>
-    Vector3<ReturnLengthUnit> getPosition()
-    {
-        updatePose();
-        return { m_Pose.x(), m_Pose.y(), 0_m };
-    }
-
-    template<typename ReturnAngleUnit = AngleUnit>
-    std::array<ReturnAngleUnit, 3> getAttitude()
-    {
-        updatePose();
-        return { m_Pose.yaw(), 0_rad, 0_rad };
-    }
-
-    const auto &getPose()
-    {
-        updatePose();
-        return m_Pose;
-    }
 
     const auto &getWheelSpeeds(){
-        updatePose();
-        return wheel_speeds;
+        updateSpeeds();
+        return m_WheelSpeeds;
     }
     virtual meters_per_second_t getAbsoluteMaximumSpeed() const override
     {
         return m_MaximumSpeed;
     }
 
-    void setPose(const Pose2<LengthUnit, AngleUnit> &pose)
-    {
-        m_MoveStopwatch.start();
-        m_Pose = pose;
-    }
-
     virtual void tank(float left, float right) override
     {
-        updatePose();
+        updateSpeeds();
         BOB_ASSERT(left >= -1.f && left <= 1.f);
         BOB_ASSERT(right >= -1.f && right <= 1.f);
         m_Left = left * m_MaximumSpeed;
         m_Right = right * m_MaximumSpeed;
     }
 
 private:
-    Pose2<LengthUnit, AngleUnit> m_Pose;
-    Stopwatch m_MoveStopwatch;
+
     const meters_per_second_t m_MaximumSpeed;
-    const millimeter_t m_AxisLength;
     meters_per_second_t m_Right{}, m_Left{};
-    std::pair <meters_per_second_t, meters_per_second_t> wheel_speeds;
+    std::pair <meters_per_second_t, meters_per_second_t> m_WheelSpeeds;
```

Comment:
As far as I can see, ``m_WheelSpeeds`` just contains the same information as ``m_Right`` and ``m_Left``, right? In which case you'd only need one or the other.